internal/core/src/storage/OpenDALChunkManager.cpp

```diff
@@ -0,0 +1,198 @@
+    if (op.error != nullptr) {
+        THROWOPENDALERROR(op.error, "Init opendal error");
+    }
+    op_ptr_ = op.op;
+    opendal_operator_options_free(op_options_);
+    LOG_SEGCORE_INFO_ << "init OpenDALChunkManager with parameter[storage: '"
+                      << storageType << ", " << storage_config.cloud_provider
+                      << "', endpoint: '" << storage_config.address
+                      << "', default_bucket_name:'"
+                      << storage_config.bucket_name << "', use_secure:'"
+                      << std::boolalpha << storage_config.useSSL << "']";
+}
+
+OpenDALChunkManager::~OpenDALChunkManager() {
+    opendal_operator_free(op_ptr_);
+}
+
+uint64_t
+OpenDALChunkManager::Size(const std::string& filepath) {
+    auto ret = opendal_operator_stat(op_ptr_, filepath.c_str());
+    if (ret.error != nullptr) {
+        THROWOPENDALERROR(ret.error, "GetObjectSize");
+    }
+    return opendal_metadata_content_length(ret.meta);
+}
+
+bool
+OpenDALChunkManager::Exist(const std::string& filepath) {
+    auto ret = opendal_operator_is_exist(op_ptr_, filepath.c_str());
+    if (ret.error != nullptr) {
+        THROWOPENDALERROR(ret.error, "ObjectExists");
+    }
+    return ret.is_exist;
+}
+
+void
+OpenDALChunkManager::Remove(const std::string& filepath) {
+    auto ret = opendal_operator_delete(op_ptr_, filepath.c_str());
+    if (ret != nullptr) {
+        THROWOPENDALERROR(ret, "RemoveObject");
+    }
+}
+
+std::vector<std::string>
+OpenDALChunkManager::ListWithPrefix(const std::string& filepath) {
+    auto ret = opendal_operator_list(op_ptr_, filepath.c_str());
+    if (ret.error != nullptr) {
+        THROWOPENDALERROR(ret.error, "ListObjects");
+    }
+    auto lister = ret.lister;
+    std::vector<std::string> objects;
+    opendal_result_lister_next result = opendal_lister_next(lister);
+    if (result.error != nullptr) {
+        opendal_lister_free(lister);
+        THROWOPENDALERROR(result.error, "ListObjects");
+    }
+    auto entry = result.entry;
+    while (entry) {
+        const char* de_path = opendal_entry_path(entry);
+        objects.push_back(std::string(de_path));
+        opendal_entry_free(entry);
+        result = opendal_lister_next(lister);
+        if (result.error != nullptr) {
+            opendal_lister_free(lister);
+            THROWOPENDALERROR(result.error, "ListObjects");
+        }
+        entry = result.entry;
+    }
+    opendal_lister_free(lister);
+    return objects;
+}
+
+uint64_t
+OpenDALChunkManager::Read(const std::string& filepath,
+                          void* buf,
+                          uint64_t size) {
+    auto ret = opendal_operator_reader(op_ptr_, filepath.c_str());
+    if (ret.error != nullptr) {
+        THROWOPENDALERROR(ret.error, "GetObjectBuffer");
+    }
+    uint64_t buf_size = 16 * 1024;
+    uint64_t buf_index = 0;
+    while (true) {
+        auto read_ret = opendal_reader_read(
```

Comment:
is there a reason we need to do it in 16K buf?
this may greatly reduce read performance.